File: app/src/main/java/com/jpcn/chatapp/ui/activities/StartActivity.java
package com.jpcn.chatapp.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.jpcn.chatapp.R;

public class StartActivity extends AppCompatActivity {

    Button login;
    Button register;

    FirebaseUser firebaseUser;

    @Override
    protected void onStart() {
        super.onStart();
        firebaseUser = FirebaseAuth.getInstance().getCurrentUser();

        if (firebaseUser != null) {
            Intent intent = new Intent(StartActivity.this, MainActivity.class);
            startActivity(intent);
            finish();
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);

        login = findViewById(R.id.buttonLogin);
        register = findViewById(R.id.buttonRegister);


        login.setOnClickListener(v -> startActivity(new Intent(StartActivity.this, LoginActivity.class)));
        register.setOnClickListener(v -> startActivity(new Intent(StartActivity.this, RegisterActivity.class)));
    }
}

File: app/src/main/java/com/jpcn/chatapp/ui/activities/StartActivity.kt
package com.jpcn.chatapp.ui.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.jpcn.chatapp.databinding.ActivityStartBinding

class StartActivity : AppCompatActivity() {
    override fun onStart() {
        super.onStart()
        FirebaseAuth.getInstance().currentUser?.let {
            val intent = Intent(this@StartActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityStartBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.buttonLogin.setOnClickListener {
            startActivity(
                Intent(this@StartActivity, LoginActivity::class.java)
            )
        }
        binding.buttonRegister.setOnClickListener {
            startActivity(
                Intent(this@StartActivity, RegisterActivity::class.java)
            )
        }
    }
}